src/clientmgr.c

```diff
@@ -13,37 +13,33 @@
 static bool client_is_active(const struct client *client);
 static void checkclient_task(void *d);
 static void checkclient(clientmgr_ctx *ctx, uint8_t mac[6]);
-static const char *state_str(enum ip_state state);
-
-struct in6_addr node_client_ip_from_mac(uint8_t mac[6]) {
-	struct in6_addr address = {};
-	inet_pton(AF_INET6, NODE_CLIENT_PREFIX, &address);
-
-	address.s6_addr[8] = mac[0] ^ 0x02;
-	address.s6_addr[9] = mac[1];
-	address.s6_addr[10] = mac[2];
-	address.s6_addr[11] = 0xff;
-	address.s6_addr[12] = 0xfe;
-	address.s6_addr[13] = mac[3];
-	address.s6_addr[14] = mac[4];
-	address.s6_addr[15] = mac[5];
-
-	return address;
+static const char *state_str(enum ipaddress_state state);
+
+struct in6_addr mac2ipv6(uint8_t mac[6]) {
+	struct in6_addr ipv6_address = {};
+	inet_pton(AF_INET6, NODE_CLIENT_PREFIX, &ipv6_address);
+
+	ipv6_address.s6_addr[8] = mac[0] ^ 0x02;
+	ipv6_address.s6_addr[9] = mac[1];
+	ipv6_address.s6_addr[10] = mac[2];
+	ipv6_address.s6_addr[11] = 0xff;
+	ipv6_address.s6_addr[12] = 0xfe;
+	ipv6_address.s6_addr[13] = mac[3];
+	ipv6_address.s6_addr[14] = mac[4];
+	ipv6_address.s6_addr[15] = mac[5];
+
+	return ipv6_address;
 }
 
 bool prefix_contains(const struct prefix *prefix, struct in6_addr *addr) {
-	int plen = prefix->plen;
 
-	for (int i = 0; i < 16; i++) {
-		int mask = ~((1<<(8 - (plen > 8 ? 8 : plen))) - 1);
+	for (int remaining_prefix_length = prefix->length, i=0;remaining_prefix_length > 0; remaining_prefix_length-= 8) {
```

Comment:
will do. when re-formatting I will keep it in there as prefix_contains_v2 just to make refactoring and renaming things easier but it will be omitted in the prettify-commit itself.